pkg/build/gobuild.go

```diff
@@ -335,14 +335,20 @@ func buildEnv(platform v1.Platform, userEnv, configEnv []string) ([]string, erro
 		"GOOS=" + platform.OS,
 		"GOARCH=" + platform.Architecture,
 	}
-
-	if strings.HasPrefix(platform.Architecture, "arm") && platform.Variant != "" {
-		goarm, err := getGoarm(platform)
-		if err != nil {
-			return nil, fmt.Errorf("goarm failure: %w", err)
-		}
-		if goarm != "" {
-			env = append(env, "GOARM="+goarm)
+	if platform.Variant != "" {
+		switch platform.Architecture {
+		case "arm":
```

Comment:
This drops arm64.

Reply:
This is intentional.

> This change also changes behavior of GOARM, since docs suggest that it's only used when the GOARCH=arm, and not arm64.

The docs don't seem to mention GOARM's behavior when GOARCH=arm64, only when GOARCH=arm. Maybe I'm misreading?

Reply:
Well there it is.